test/patterns/partially_synchronized_methods/FullSynchronization.java

```diff
@@ -0,0 +1,15 @@
+public class SingleClass {
+    int synchronizationField = 0;
+
+    public SingleClass() {
+        synchronized(synchronizationField) {
+            int x = 0;
+        }
+    }
+
+    public void method() {
+        synchronized(synchronizationField) {
```

Comment:
can you add in this case different type of synchronization? We had a lot of test when we had Class inside, array, function, etc.

Reply:
There is no point in it. Current pattern do nothing with class, functions and nested scopes. So it really does not matter what is inside `synchronized` block, it all ignored.